src/node.js

```diff
@@ -220,19 +220,22 @@ class Node {
           const str = String(data).trim()
           const match = str.match(/API server listening on (.*)/)
           const gwmatch = str.match(/Gateway (.*) listening on (.*)/)
+          const readyMatch = str.match(/Daemon is ready/)
 
           if (match) {
             this._apiAddr = multiaddr(match[1])
             this.api = ipfs(match[1])
             this.api.apiHost = this.apiAddr.nodeAddress().address
             this.api.apiPort = this.apiAddr.nodeAddress().port
+          }
 
-            if (gwmatch) {
-              this._gatewayAddr = multiaddr(gwmatch[2])
-              this.api.gatewayHost = this.gatewayAddr.nodeAddress().address
-              this.api.gatewayPort = this.gatewayAddr.nodeAddress().port
-            }
+          if (gwmatch) {
+            this._gatewayAddr = multiaddr(gwmatch[2])
+            this.api.gatewayHost = this.gatewayAddr.nodeAddress().address
+            this.api.gatewayPort = this.gatewayAddr.nodeAddress().port
+          }
 
+          if (readyMatch) {
             callback(null, this.api)
           }
```

Comment:
Yeah, this part was getting called multiple times (twice in this case), first with "API listening..." and then with "Gateway (readonly) listening...\nDaemon is ready". Not sure why or if that's the intended behaviour, but adding an else and an error callback will just make it fail always (from user's perspective).

@dignifiedquire I believe you refactored the exec code last time, any ideas here?